walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.lang.Math;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> reqAttendees = request.getAttendees();
+    Collection<String> allAttendees = new HashSet<String>(){{
+        addAll(request.getAttendees());
+        addAll(request.getOptionalAttendees());
+    }};
+    ArrayList<TimeRange> finalTimes = determineFreeTimes(allAttendees, events, request);
+    if (finalTimes.isEmpty()){
+        finalTimes = determineFreeTimes(reqAttendees, events, request);
+    }
+    return finalTimes;
+  }
+
+  private ArrayList<TimeRange> determineFreeTimes(Collection<String> attendees, Collection<Event> events, MeetingRequest request){
```

Comment:
Could you please add Javadoc/comments for each method? https://google.github.io/styleguide/javaguide.html#s7-javadoc